// @zaeem/types/src/order.ts
export interface Table {
  id: string;
  number: number;
  capacity: number;
  branchId: string;
  qrCode: string;
  status: "free" | "occupied" | "reserved" | "cleaning";
}

export interface OrderModifier {
  id: string;
  name: string;
  priceCents: number;
}

export interface OrderItem {
  id: string;
  menuItemId: string;
  name: string;
  quantity: number;
  unitPriceCents: number;
  modifiers: OrderModifier[];
  notes: string;
}

export type PaymentMethod = "cash" | "card" | "wallet" | "credit";

export interface Payment {
  id: string;
  method: PaymentMethod;
  amountCents: number;
  tipCents: number;
  paidAt: string;
}

export type OrderStatus =
  | "draft"
  | "pending"
  | "preparing"
  | "ready"
  | "served"
  | "paid"
  | "cancelled";

export interface Order {
  id: string;
  tableId: string;
  branchId: string;
  items: OrderItem[];
  payments: Payment[];
  status: OrderStatus;
  subtotalCents: number;
  taxCents: number;
  totalCents: number;
  createdAt: string;
  updatedAt: string;
}
