// @zaeem/types/tailwind-preset.js
/** @type {import('tailwindcss').Config} */
const preset = {
  darkMode: "class",
  theme: {
    extend: {
      colors: {
        canvas: "#F8FAFC",
        surface: "#FFFFFF",
        sidebar: "#FFFFFF",
        cart: "#FAFAFA",
        primary: {
          50: "#ECFDF5",
          100: "#D1FAE5",
          200: "#A7F3D0",
          300: "#6EE7B7",
          400: "#34D399",
          500: "#10B981",
          600: "#059669",
          700: "#047857",
          800: "#065F46",
          900: "#064E3B",
        },
        slate: {
          50: "#F8FAFC",
          100: "#F1F5F9",
          200: "#E2E8F0",
          300: "#CBD5E1",
          400: "#94A3B8",
          500: "#64748B",
          600: "#475569",
          700: "#334155",
          800: "#1E293B",
          900: "#0F172A",
        },
        success: "#10B981",
        warning: "#F59E0B",
        danger: "#EF4444",
        info: "#3B82F6",
        pos: {
          bg: "#F8FAFC",
          card: "#FFFFFF",
          sidebar: "#FFFFFF",
          cart: "#FAFAFA",
          border: "#E2E8F0",
          text: "#0F172A",
          muted: "#64748B",
          accent: "#059669",
          dangerBg: "#FEF2F2",
          warningBg: "#FFFBEB",
          successBg: "#ECFDF5",
        },
      },
      fontFamily: {
        sans: ["Inter", "IBM Plex Sans Arabic", "system-ui", "sans-serif"],
        display: ["Inter", "sans-serif"],
        arabic: ["IBM Plex Sans Arabic", "sans-serif"],
        mono: ["IBM Plex Mono", "monospace"],
      },
      borderRadius: {
        sm: "8px",
        DEFAULT: "10px",
        md: "12px",
        lg: "16px",
        xl: "20px",
        "2xl": "24px",
      },
      boxShadow: {
        card: "0 0 0 1px rgba(0,0,0,0.03), 0 1px 2px rgba(0,0,0,0.04), 0 2px 4px rgba(0,0,0,0.02)",
        "card-hover":
          "0 0 0 1px rgba(0,0,0,0.04), 0 4px 6px rgba(0,0,0,0.04), 0 10px 15px rgba(0,0,0,0.03)",
        elevated:
          "0 0 0 1px rgba(0,0,0,0.05), 0 10px 15px rgba(0,0,0,0.05), 0 20px 25px rgba(0,0,0,0.03)",
      },
    },
  },
  plugins: [],
};

module.exports = preset;
