// @zaeem/types/tailwind-preset.js
/** @type {import('tailwindcss').Config} */
const preset = {
  darkMode: "class",
  theme: {
    extend: {
      colors: {
        canvas: "var(--bg)",
        surface: "var(--surface)",
        "surface-alt": "var(--surface-alt)",
        text: {
          DEFAULT: "var(--text)",
          2: "var(--text-2)",
          3: "var(--text-3)",
          muted: "var(--text-muted)",
        },
        line: {
          DEFAULT: "var(--line)",
          2: "var(--line-2)",
        },
        accent: {
          DEFAULT: "var(--accent)",
          soft: "var(--accent-soft)",
          text: "var(--accent-text)",
        },
        ok: "var(--ok)",
        warn: "var(--warn)",
        danger: {
          DEFAULT: "var(--danger)",
          soft: "var(--danger-soft)",
        },
        ink: {
          50: "var(--ink-50)",
          100: "var(--ink-100)",
          200: "var(--ink-200)",
          300: "var(--ink-300)",
          400: "var(--ink-400)",
          500: "var(--ink-500)",
          600: "var(--ink-600)",
          700: "var(--ink-700)",
          800: "var(--ink-800)",
          900: "var(--ink-900)",
        },
        // Root-cause fix: these were referenced by bg-saffron-*/bg-live-*
        // etc. across the app but never defined here -- Tailwind emitted no
        // CSS for them at all, which is why buttons using them had no fill
        // (white/transparent) under white text. See index.css for the
        // ramp values and why 600/700/800 are darker than raw --accent.
        saffron: {
          50: "var(--saffron-50)",
          100: "var(--saffron-100)",
          200: "var(--saffron-200)",
          300: "var(--saffron-300)",
          400: "var(--saffron-400)",
          500: "var(--saffron-500)",
          600: "var(--saffron-600)",
          700: "var(--saffron-700)",
          800: "var(--saffron-800)",
          900: "var(--saffron-900)",
        },
        live: {
          50: "var(--live-50)",
          100: "var(--live-100)",
          200: "var(--live-200)",
          300: "var(--live-300)",
          400: "var(--live-400)",
          500: "var(--live-500)",
          600: "var(--live-600)",
          700: "var(--live-700)",
          800: "var(--live-800)",
          900: "var(--live-900)",
        },
      },
      fontFamily: {
        sans: ["Tajawal", "system-ui", "sans-serif"],
        arabic: ["Tajawal", "sans-serif"],
        mono: ["IBM Plex Mono", "monospace"],
      },
      borderRadius: {
        sm: "7px",
        DEFAULT: "10px",
        md: "10px",
        lg: "13px",
        xl: "13px",
        "2xl": "13px",
      },
      boxShadow: {
        "sh-1": "0 1px 2px rgba(16,24,40,.05)",
        "sh-2": "0 1px 2px rgba(16,24,40,.05), 0 6px 16px rgba(16,24,40,.04)",
        "sh-3": "0 1px 2px rgba(16,24,40,.05), 0 8px 24px rgba(16,24,40,.05)",
      },
    },
  },
  plugins: [],
};

module.exports = preset;
